1. Маємо результати A/B-тесту для інтернет-магазину. Вибірка містить інформацію про відсоток конверсій для двох груп:

- клієнти, які використовують стару програму лояльності (група A);
- клієнти, які використовують нову програму лояльності (група B).

Завдання — проаналізувати конверсії та порівняти ефективність нової програми лояльності із попередньою.

In [2]:
import pandas as pd
from scipy.stats import shapiro, ttest_ind


# given data
conversion_rate_A = [5.2, 4.8, 6.1, 5.0, 5.3, 4.9, 5.1, 5.4, 4.7, 5.0]
conversion_rate_B = [6.3, 6.7, 6.9, 6.2, 7.0, 6.5, 6.8, 6.6, 6.1, 6.4]

data = {
    "conversion_rate_A": [5.2, 4.8, 6.1, 5.0, 5.3, 4.9, 5.1, 5.4, 4.7, 5.0],
    "conversion_rate_B": [6.3, 6.7, 6.9, 6.2, 7.0, 6.5, 6.8, 6.6, 6.1, 6.4]
}

df = pd.DataFrame(data)


# Normality test (determine whether the data distribution is normal)
_, p_value_A = shapiro(df["conversion_rate_A"])
_, p_value_B = shapiro(df["conversion_rate_B"])

print(f"p-value of normality test for conversion_rate_A: {p_value_A}")
print(f"p-value of normality test for conversion_rate_B: {p_value_B}")


# Perform t-test
t_stat, p_value = ttest_ind(df["conversion_rate_A"], df["conversion_rate_B"])

print(f"t-statistic: {t_stat}, p-value: {p_value}")

p-value of normality test for conversion_rate_A: 0.10622253110873986
p-value of normality test for conversion_rate_B: 0.8923673061903017
t-statistic: -8.85437744847146, p-value: 5.611140642425496e-08


Спочатку перевірили обидва набори даних на нормальність розподілу:\
p-value of normality test for conversion_rate_A: 0.10622253110873986\
p-value of normality test for conversion_rate_B: 0.8923673061903017\
**p-value > 5% що свідчить про те, що обидва набори даних мають нормальний розподіл**

Далі використали вбудовану функцію **ttest_ind()** модуля stats бібліотеки SciPy.

Як результат,за допомогою p-value оцінюємо статистичну значущість результатів.

В результаті отримали значення **p-value=5.611140642425496e-08, що менше 0.05, а це означає, що групи А та В мають статистично значущу різницю в конверсіях**.

 **Оскільки середня конверсія для групи В — вище, то нова програма лояльності дійсно призводить до значущих ефектів.**


2. У процесі тестування нової програми лояльності також збиралися дані й про рівень задоволеності клієнтів. Маємо дві групи:

група А - рівень задоволеності по шкалі від 1 до 5 для старої програми лояльності;\
група В - для нової програми лояльності.

Завдання:
Порівняти програми лояльності за цим критерієм. Чи є нова програма ефективнішою?\
satisfaction_A = [3, 4, 5, 3, 4, 3, 4, 5, 3, 4]\
satisfaction_B = [4, 5, 5, 4, 5, 5, 4, 5, 5, 4]



In [3]:
from scipy.stats import mannwhitneyu


# given data
satisfaction_A = [3, 4, 5, 3, 4, 3, 4, 5, 3, 4]
satisfaction_B = [4, 5, 5, 4, 5, 5, 4, 5, 5, 4]

# perform Mann-Whitney U test
u_stat, p_value = mannwhitneyu(satisfaction_A, satisfaction_B)

print(f"U-statistic: {u_stat}, p_value: {p_value}")

U-statistic: 22.0, p_value: 0.025506745374925463


Перевірку на нормальність даних тут не проводимо, тому що апріорі дані мають ранговий (дискретний) характер.\
Проводимо тест Манна-Уітні за допомогою вбудованої функції mannwhitneyu().

**Отримали p-value = 0.0255 < 0.05, а це означає, що рівень задоволеності між групами А та В має статистично значущу різницю.
Рівень задоволеності клієнтів за новою програмою лояльності (satisfaction_B) є вищим, а отже нова програма лояльності дійсно призводить до значущих ефектів**.

**U-statistic: 22.0, p_value: 0.025506745374925463**

Середнє арифметичне для satisfaction_A = 3.8

Середнє арифметичне для satisfaction_B = 4.6